packages/block-editor/src/components/colors-gradients/control.js

```diff
@@ -58,10 +57,11 @@ function ColorGradientControlInner( {
 	headingLevel,
 } ) {
 	const canChooseAColor =
-		onColorChange && ( ! isEmpty( colors ) || ! disableCustomColors );
+		onColorChange &&
+		( ( colors && colors.length > 0 ) || ! disableCustomColors );
 	const canChooseAGradient =
 		onGradientChange &&
-		( ! isEmpty( gradients ) || ! disableCustomGradients );
+		( ( gradients && gradients.length > 0 ) || ! disableCustomGradients );
```

Comment:
Gradients are expected to be an array in both tests and `theme.json`:

https://github.com/WordPress/gutenberg/blob/553a2c2fecc42947ec0b9b5a87440fd1017cbbde/packages/block-editor/src/components/colors-gradients/test/control.js#L23

https://developer.wordpress.org/themes/advanced-topics/theme-json/#color-palette